code/game/objects/items/devices/radio/radio.dm

```diff
@@ -65,6 +64,9 @@
 	var/analog_secured = list() // list of accesses used for encrypted analog, mainly for mercs/raiders
 	var/datum/radio_frequency/analog_radio_connection
 
+/obj/item/radio/setup_power_supply(loaded_cell_type, accepted_cell_type, power_supply_extension_type, charge_value)
+	return ..(/obj/item/cell/device, /obj/item/cell/device, /datum/extension/loaded_cell)
```

Comment:
This will mean no radio subtype could change these things. It also clobbers/ignores the `charge_value` parameter. `..()` on its own implies all the arguments, but passing any arguments at all to `..()` means it _only_ passes those.
![image](https://github.com/NebulaSS13/Nebula/assets/110272328/4a1b15bb-686a-4397-9953-40c41e0d12a2)

Reply:
`return ..(loaded_cell_type || /obj/item/cell/device/standard, accepted_cell_type || /obj/item/cell/device, power_supply_extension_type, charge_value)` is the really unwieldy and verbose way to avoid all of those issues.
Also, use `/obj/item/cell/device/standard` for loaded_cell_type instead of `/obj/item/cell/device`. The latter should probably be an abstract type.